crypto/digestible/derive/src/lib.rs

```diff
@@ -71,8 +65,10 @@ fn try_digestible(input: TokenStream) -> Result<TokenStream, &'static str> {
         .collect::<Vec<_>>();
 
     // Final expanded result
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
     let expanded = quote! {
-        impl mc_crypto_digestible::Digestible for #ident {
+        impl #impl_generics mc_crypto_digestible::Digestible for #ident #ty_generics #where_clause {
             fn digest<D: mc_crypto_digestible::Digest>(&self, hasher: &mut D) {
                 hasher.input(stringify!(#ident).as_bytes());
```

Comment:
I'm pretty sure you want to also inject the full path to each generic here as well, otherwise you can get hash collisions between, e.g. `FlyingKite<Thing1>` and `FlyingKite<Thing2>`.